### DISCUSSION TITLE
@name expansion not working

### DISCUSSION THREAD

**Discussion by radeknoone** [element-hq/element-meta#2216] [Product Feedback]
### Steps to reproduce

I have to write person name starting with @ and wait and click on the 'graphical representation' of that name to let chat understand, that it is person not the plaintext:
![image](https://github.com/vector-im/element-web/assets/23701185/99180f2c-0e8e-4295-9c4b-432d600c148c)

This is 'graphical representation' of user:
![image](https://github.com/vector-im/element-web/assets/23701185/b710ea8c-b44d-49a9-b93d-26bd8942dfcb)

This is what happen, when I use text form: 
![image](https://github.com/vector-im/element-web/assets/23701185/bffd78c1-a2dc-4ff9-ac6b-9b75d8582d91)

Thats not making sense to me - the name IS @user:server.tld, why I have to bother with the 'graphical representation'?

Element version: 1.11.30
Olm version: 3.2.12

### Outcome

I would expect, that if i write full user name like @user:server.tld, then will be used same way as "graphical representation".

### Operating system

Linux

### Browser information

Firefox

### Will you send logs?

Yes

**Comment by t3chguy** [comment]
Because the mention has to be listed in `m.mentions` property in the event, and as a sender you may wish to send something that looks like an MXID which doesn't mention the owner of that MXID.

**Reply by radeknoone** [reply]
Hi t3chguy,

I don't understand your note. Is there anything I can do for helping with my issue?

**Reply by t3chguy** [reply]
The way you describe it working is as intended, hence I have moved your "issue" into discussions where you can propose alternatives to how it works. I explained that it is intentional the sender has to explicitly select to ping a user, this is part of a project called Intentional Mentions.

**Reply by radeknoone** [reply]
Thank you for explaining that!

**Comment by radeknoone** [comment]
In my opinion, this functionality is not correct for two reasons.

Firstly, it requires a sufficiently fast and low latency connectivity to the server, otherwise there will be no expansion and graphical interpretation options available.

Secondly, it is an unnecessary step that complicates usage and forces users to take additional steps. It is annoying even more if you are used to work without mouse.

Is there any possibility to allow the "expansion" of certain types of text (@names, :emoji: and so on) in a different way that does not require this step? Just writing it down, not to have choose from options.

**Reply by t3chguy** [reply]
> Firstly, it requires a sufficiently fast and low latency connectivity to the server, otherwise there will be no expansion and graphical interpretation options available.

This is false, the completions are done entirely locally with 0 outbound requests other than for avatars, unlike other platforms which perform them server-side.
